### ISSUE TITLE
Deprecate the copycss attribute for future removal

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor#3936]
I've been thinking a lot about this attribute and in my opinion it blurs the line between a converter and a publishing tool.

In itself it's not a big deal but then users might expect additional features. For instance, if Asciidoctor can copy a style sheet to the output directory, why can't it copy images, icons or videos to the output directory?
Similarly, a `SyntaxHighlighter` can write the associated style sheet to the output directory. But how do I write a JavaScript file used in my custom extension to the output directory?

It's certainly well intentioned but the current implementation has some limitations and I think it would be better to draw a clear line.

In short, Asciidoctor should not copy files around. That should be left up to a build/publish tool like Gradle, Maven, or Antora. In other words, if you decide to use a destination (output) directory then you will need to take care of copying all the related resources.

It will also make easier to implement https://github.com/asciidoctor/asciidoctor/issues/3471. In the future it might also ease the integration of an abstraction layer/extension point to share the location of all the related resources in the publishing pipeline.
